store/copr/coprocessor.go

```diff
@@ -244,6 +264,24 @@ func buildCopTasks(bo *Backoffer, cache *RegionCache, ranges *KeyRanges, req *kv
 		}
 		for i := 0; i < rLen; {
 			nextI := mathutil.Min(i+rangesPerTask, rLen)
+			hint := -1
+			// calculate the row count hint
+			if req.FixedRowCountHint != nil {
+				startKey, endKey := loc.Ranges.At(i).StartKey, loc.Ranges.At(nextI-1).EndKey
+				// move to the previous range if startKey of current range is lower than endKey of previous location.
```

Comment:
Better to add an example here as this part itself is quite complex and difficult to understand.